### DISCUSSION TITLE
Modules requests

### DISCUSSION THREAD

**Discussion by fannheyward** [nvim-mini/mini.nvim#53] [Ideas]
Thanks for your mini.nvim, I really like the feature-richness design, here're some modules I think can be made into mini.nvim.

- `mini.cycle` likes https://github.com/zef/vim-cycle
- `mini.lastplace` likes https://github.com/farmergreg/vim-lastplace or https://github.com/ethanholz/nvim-lastplace

**Comment by echasnovski** [comment]
Thank you for your kind words and useful suggestions!

- I already planned something along the lines of 'zef/vim-cycle'. Although, it is quite far in my backlog at the moment.
- I have never heard about 'farmergreg/vim-lastplace'. Although I didn't quite get the purpose of plugin from reading README (three times), the source code seems to be quite small. Not really sure it is worth extra module, to be honest. Some day, maybe a function in 'mini.misc'.

**Reply by fannheyward** [reply]
`vim-lastplace` will remember your last cursor position on a document, when you open it again next time, it will restore your position to the place.

**Comment by xigoi** [comment]
Instead of `lastplace`, simply use this VimScript snippet, which is given somewhere in the Vim documentation:

```vim
autocmd BufReadPost * if line("'\"") >= 1 && line("'\"") <= line("$") | exe "normal! g`\"" | endif
```

**Comment by cryptomilk** [comment]
Support for restoring the cursor has been added to mini.misc with https://github.com/echasnovski/mini.nvim/commit/5be4676b9744b7a59054961d00d6197ef9e68186
